contracts/contracts/Subcall.sol

```diff
@@ -182,11 +188,39 @@ library Subcall {
         pure
         returns (uint256 newOffset, uint256 value)
     {
-        if (result[offset] & 0x40 != 0x40) revert InvalidLength();
+        uint8 prefix = uint8(result[offset]);
+        uint256 len;
 
-        uint256 len = uint8(result[offset++]) ^ 0x40;
+        if (prefix <= 0x17) {
+            return (offset + 1, prefix);
+        }
+        // Byte array, parsed as a big-endian integer.
+        else if (prefix & 0x40 == 0x40) {
+            len = uint8(result[offset++]) ^ 0x40;
+        }
+        // Unsigned integer, CBOR encoded.
+        else if (prefix & 0x10 == 0x10) {
+            if (prefix == 0x18) {
+                len = 1;
+            } else if (prefix == 0x19) {
+                len = 2;
+            } else if (prefix == 0x1a) {
+                len = 4;
+            } else if (prefix == 0x1b) {
+                len = 8;
+            } else if (prefix == 0x1d) {
+                len = 16;
+            } else {
```

Comment:
I see now, that this is actually wrong https://www.rfc-editor.org/rfc/rfc8949.html#name-jump-table-for-initial-byte